### PR TITLE
bigtable: throw when required information is missing

### PR DESCRIPTION
**Pull request by stephenplusplus**
In Bigtable, we should throw whenever data that is required is not provided. I tried to find those places, but @callmehiphop, could you take a look and let me know if I missed any?
#### To Dos
- [x] Throw
  - [x] Bigtable
  - [x] Bigtable#createTable
  - [x] row#createRules
  - [x] table#createFamily
  - [x] table#family
  - [x] table#row
- [x] Unit tests

### CODE REVIEW

**Comment by callmehiphop** [comment]
Can you clarify what kind of data you want to throw for? For example, in `Row#increment` we don't throw if a user were to provide a bad type for the `column` value - your changes make me think you only want to throw an error where grpc/upstream errors can't give back a detailed message of what is wrong.

**Comment by stephenplusplus** [comment]
> Can you clarify what kind of data you want to throw for?

Somewhere between "let the API handle it all" and "validate all user input". In places like "create____", we know that it's impossible to create a "____" without a name. So, we should blow up and let the user know.

We especially should throw in all cases where our code assumes we have the input and tries to manipulate it, e.g., "create____" takes a string, but it needs to be uppercased, so our code throws on the line we tried to use their input, instead of throwing that their input was wrong and how to fix it.

With `increment`, it's a judgement call. What do you think?

> your changes make me think you only want to throw an error where grpc/upstream errors can't give back a detailed message of what is wrong.

Not exactly, but where possible, the API should do the error reporting for us. I do want to catch completely obvious things, basically falling under the label of "you used *OUR* API wrong", so not things like "does the name match some regex that is enforced by the API".

**Comment by callmehiphop** [comment]
Thanks for that. I think some places you'd definitely want to look at are

* `Row` constructor - `name` param. 
* `Family` constructor - `name` param. 

Possible maybes are

* `Row#createRules` - `rules` param.
* `Row#filter` - `filter` param.
* `Row#deleteCells` - `columns` param.
* `Row#increment` - `column` param.
* `Row#save` - `key` param.
* `Family#setMetadata` - `metadata` param.
* `Table#insert` - `entries` param.
* `Table#mutate` - `entries` param.

File: packages/bigtable/src/index.js [hunk 1]
```diff
@@ -409,6 +422,10 @@ Bigtable.prototype.createTable = function(name, options, callback) {
     options = {};
   }
 
+  if (!name) {
```
Comment (callmehiphop) [inline packages/bigtable/src/index.js]: I believe this is currently untested.

Comment (stephenplusplus) [inline packages/bigtable/src/index.js]: Woops. Added!

**Comment by callmehiphop** [comment]
LGTM, waiting for CI.

File: packages/bigtable/test/index.js [hunk 2]
```diff
@@ -106,12 +106,27 @@ describe('Bigtable', function() {
       fakeUtil.normalizeArguments = normalizeArguments;
     });
 
+    it('should throw if a cluster is not provided', function() {
+      assert.throws(function() {
+        new Bigtable();
```
Comment (callmehiphop) [inline packages/bigtable/test/index.js]: Not sure why, but coveralls is claiming this is untested. Could we try changing this to:

```js
new Bigtable({
  zone: ZONE
});
```

Comment (callmehiphop) [inline packages/bigtable/test/index.js]: https://coveralls.io/builds/7328412/source?filename=packages%2Fbigtable%2Fsrc%2Findex.js#L285

Comment (stephenplusplus) [inline packages/bigtable/test/index.js]: Okidoke, fixed.

**Comment by stephenplusplus** [comment]
I snuck in a change to add a `covers` task as well here.

**Pull request merged**